cli/pcluster/config/param_types.py

```diff
@@ -0,0 +1,1202 @@
+        cfn_converter = self.map.get("cfn", None)
+        if cfn_converter and cfn_params:
+            # initialize the value from cfn or from map according to the scheduler
+            if get_cfn_param(cfn_params, "Scheduler") == "awsbatch":
+                self._init_from_map()
+            else:
+                # MinSize param > 0 means that maintain_initial_size was set to true at cluster creation
+                min_size_cfn_value = get_cfn_param(cfn_params, cfn_converter) if cfn_converter else "0"
+                min_size_value = int(min_size_cfn_value) if min_size_cfn_value != "NONE" else 0
+                self.value = min_size_value > 0
+        elif cfn_value:
+            self.value = self.get_value_from_string(cfn_value)
+        else:
+            self._init_from_map()
+
+    def to_cfn(self):
+        """Convert parameter to CFN representation."""
+        cfn_params = {}
+
+        cluster_config = self.pcluster_config.get_section(self.section_key)
+        if cluster_config.get_param_value("scheduler") != "awsbatch":
+            cfn_value = cluster_config.get_param_value("maintain_initial_size")
+            min_size_value = cluster_config.get_param_value("initial_queue_size") if cfn_value else "0"
+            cfn_params.update({self.map.get("cfn"): str(min_size_value)})
+
+        return cfn_params
+
+
+class MinSizeParam(IntParam):
+    """
+    Class to manage the min_vcpus configuration parameters.
+
+    We need this class since the same CFN input parameter "MinSize" is populated
+    from the "min_vcpus" parameter when the scheduler is awsbatch and
+    merging info from "initial_queue_size" and "maintain_initial_size" when the scheduler is a traditional one.
+    """
+
+    def _init_from_cfn(self, cfn_params=None, cfn_value=None):
+        """Initialize param value by parsing the right CFN input according to the scheduler."""
+        cfn_converter = self.map.get("cfn", None)
+        if cfn_converter and cfn_params:
+            cfn_value = get_cfn_param(cfn_params, cfn_converter) if cfn_converter else "NONE"
+            # initialize the value from cfn or from map according to the scheduler
+            if get_cfn_param(cfn_params, "Scheduler") == "awsbatch":
+                self.value = self.get_value_from_string(cfn_value)
+            else:
+                self._init_from_map()
+        elif cfn_value:
+            self.value = self.get_value_from_string(cfn_value)
+        else:
+            self._init_from_map()
+
+    def to_cfn(self):
+        """Convert parameter to CFN representation."""
+        cfn_params = {}
+
+        cluster_config = self.pcluster_config.get_section(self.section_key)
+        if cluster_config.get_param_value("scheduler") == "awsbatch":
+            cfn_value = cluster_config.get_param_value("min_vcpus")
+            cfn_params.update({self.map.get("cfn"): str(cfn_value)})
+
+        return cfn_params
+
+
+class AvailabilityZoneParam(Param):
```

Comment:
The master_availability_zone is a `Param` object in the vpc `Section`. 
It means that the value can be retrieved from the section with `get_param_value`, etc.

If we remove it from the mapping, we have to create a `MasterSubnetIdParam` to manage `from_file/from_cfn/to_cfn` conversion for both `master_subnet_id` and `master_availability_zone` parameters (3 functions more complex than the `from_file` we have so far).

Furthermore, we have to create the Param object for the avail zone in the `from_file` initialization of the master subnet id and it would be the single case in all the code on which a Param create another Param and add it to the PclusterConfig.

I'd leave it as is for now since it is enough simple and clear.